### ISSUE TITLE
Rendering issue in Castlevania: Dawn of Sorrow

### ISSUE THREAD

**Issue by windwakr** [TASEmulators/desmume#660]
This boss in Catlevania Dawn of Sorrow(game code NTR-ACVE-USA) an issue where the overlap of the arm segments near his elbow will flash black.
![236331261-87d0df0e-3f58-4413-89c5-d1a4cad33250](https://user-images.githubusercontent.com/284886/236335817-b0e7313e-4408-4f5f-9406-569c4a563473.png)
Affects both the software renderer and OpenGL renderer. I've verified that it does not happen on real hardware.

Here is a save file right before him. He's to the left of the save room.
[Castlevania.-.Dawn.of.Sorrow.zip](https://github.com/TASEmulators/desmume/files/11401779/Castlevania.-.Dawn.of.Sorrow.zip)

**Comment by zeromus** [comment]
Can you find a capture from hardware of this game at this point? This seems like a strange thing for us to be getting wrong. It's a complex game. I'd say there's a chance it's bugged in the original game. Not a big chance, but.. a chance.

**Comment by windwakr** [comment]
Here's a vid I took. Ignore the tiny blinking red spot near the center, it's a reflection from my phone.
[vid.webm](https://github.com/TASEmulators/desmume/assets/284886/ab50cbb0-b11f-4a94-ad11-f96be5566135)

I've also found that old versions of DeSmuME don't have this issue.

**Comment by zeromus** [comment]
ok, well done. thanks.

**Comment by rogerman** [comment]
@windwakr: Can you confirm which version of DeSmuME didn't have this issue? I tried all the way back to v0.9.8 and I still saw the graphical issue. Also, for the version that did work, were you running SoftRasterizer or OpenGL?

**Comment by windwakr** [comment]
0.9.6 seems to be the most recent release that doesn't have the flickering. Renderer doesn't matter.

**Comment by zeromus** [comment]
I dont think it has to do with the sorting. It must be that the depth values are getting disordered. But you would expect that system to be really resilient in this game.

**Comment by rofl0r** [comment]
in case the git history goes back to 0.9.6, git bisect is probably a good option to pinpoint the exact change breaking this.
